server/manager/src/main/java/org/apache/accumulo/manager/ManagerClientServiceHandler.java

```diff
@@ -434,7 +441,7 @@ public void setSystemProperty(TInfo info, TCredentials c, String property, Strin
       throw iae;
     } catch (Exception e) {
       Manager.log.error("Problem setting config property in zookeeper", e);
-      throw new TException(e.getMessage());
+      throw new ThriftPropertyException(property, value, e.getMessage());
```

Comment:
Do you want this above where IllegalArgumentException is caught?

Reply:
Yes, I moved up my Thrift property Exception to handle the caught `iae`